_rules/table-headers-attribute-refer-to-data-cells-a25f45.md

```diff
@@ -0,0 +1,231 @@
+- Tables using `headers` id associations are intended as data rather than [layout table](#layout-table).
+
+## Accessibility Support
+
+- `headers` attribute may not be consistently announced by assistive technologies.
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Information and relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
+- [H43: Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/WAI/WCAG21/Techniques/html/H43)
+- [F90: Incorrectly associating table headers and content via the headers and id attributes](https://www.w3.org/WAI/WCAG21/Techniques/failures/F90)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The `headers` attribute on the cell refers to `th` within the same `table`.
+
+```html
+<table>
+  <thead>	
+    <tr>
+      <th id="header1">Projects</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td headers="header1">15%</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+#### Passed Example 2
+
+The `headers` attribute on the cell refers to `th` within the same `table`. Multiple headers are referenced for a cell with `colspan` of `2`.
+
+```html
+<table>
+  <thead>	
+    <tr>
+      <th id="header1">Projects</th>
+      <th id="header2">Exams</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td colspan="2" headers="header1 header2">15%</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+#### Passed Example 3
+
+The `headers` attribute on the cell refers to `td` with a role of `columnheader` within the same `table`.
+
+```html
+<table>
+  <thead>	
+    <tr>
+      <td role='columnheader' id="header1">Projects</td>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td headers="header1">15%</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+#### Passed Example 4
+
+A `table` with multiple `columnheader(s)`, where the `headers` attribute on the cell lists the `id(s)` of `th` within the same `table`.
+
+```html
+<table>
+  <tr>
+    <th colspan="2" id="header1">Projects</th>
+    <th colspan="2" id="header2">Exams</th>
+  </tr>
+  <tr>
+    <th id="e1" headers="header1">1</th>
+    <th id="e2" headers="header1">2</th>
+    <th id="p1" headers="header2">1</th>
+    <th id="p2" headers="header2">2</th>
+  </tr>
+  <tr>
+    <td colspan="2" headers="header1 e1 e2">15%</td>
+    <td headers="header2 p1">15%</td>
+    <td headers="header2 p2">45%</td>
+  </tr>
+</table>
+```
+
+### Failed
+
+#### Failed Example 1
+
+Cells with `headers` attribute, refers to non-existing `th` within the same `table`.
+
+```html
+<table>
+  <tr>
+    <th id="header1">Projects</th>
+    <th id="header2">Exams</th>
+  </tr>
+  <tr>
+    <td headers="header3">15%</td>
+    <td headers="header3">15%</td>
+  </tr>
+</table>
+```
```

Comment:
I think this example would also fail an HTML validation test. Is this still something to include in an accessibility test? (just double checking)

Reply:
@EmmaJP in this case we are simply using a `header3` as an `id` that does not exist. I ran the HTML validation [`html-hint`](https://www.npmjs.com/package/htmlhint) against the above snippet and it seemed ok.

Reply:
W3 nu validator throws an error when 'header 3' does not exist, and also when 'header 3' exists as an id in a different table. So this example is also invalid HTML.

I've always thought valid HTML a prerequisite for accessibility rules.

Reply:
Right. I think we will need to fix this out of scope of this PR for 2 reasons.

- I currently have a different pr, which checks validity of HTML using `html-hint`, see https://github.com/act-rules/act-rules.github.io/pull/631
- I would like to integrate something like w3 nu validator as a part of continuous integration testing.

In this specific example we have refer to an id that does not exist, so yes it will refer to an `does not exist` error.

I have created an issue for this Emma - https://github.com/act-rules/act-rules.github.io/issues/679